### ISSUE TITLE
mach build-cef could not compile embedding

### ISSUE THREAD

**Issue by cb88** [servo/servo#7324]
I imagine this may just be a temporary problem but I can build servo itself but not ./mach build-cef.

This is on Gentoo amd64 I don't imagine that matters much as servo uses a snapshot of rust and not my natively installed version as far as I am aware.    

```
 ./mach env
export PATH=/root/servo/.servo/rust/7b7fc67dd453c470a48dbdcf64693a93293c9ab0/rustc-1.4.0-dev-x86_64-unknown-linux-gnu/rustc/bin:/root/servo/.servo/rust/7b7fc67dd453c470a48dbdcf64693a93293c9ab0/rustc-1.4.0-dev-x86_64-unknown-linux-gnu/bin:/root/servo/.servo/cargo/2015-08-20/cargo/bin:/root/servo/.servo/cargo/2015-08-20/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/bin:/usr/x86_64-pc-linux-gnu/gcc-bin/4.9.2
export LD_LIBRARY_PATH=/root/servo/.servo/rust/7b7fc67dd453c470a48dbdcf64693a93293c9ab0/rustc-1.4.0-dev-x86_64-unknown-linux-gnu/rustc/lib:/root/servo/.servo/rust/7b7fc67dd453c470a48dbdcf64693a93293c9ab0/rustc-1.4.0-dev-x86_64-unknown-linux-gnu/lib:
```
```
Compiling embedding v0.0.1 (file:///root/servo/ports/cef)
browser.rs:230:9: 244:10 error: the trait `core::iter::Iterator` is not implemented for the type `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` [E0277]
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 help: run `rustc --explain E0277` to see a detailed explanation
browser.rs:230:9: 244:10 note: `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` is not an iterator; maybe try calling `.iter()` or a similar method
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 error: the trait `core::iter::Iterator` is not implemented for the type `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` [E0277]
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 help: run `rustc --explain E0277` to see a detailed explanation
browser.rs:230:9: 244:10 note: `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` is not an iterator; maybe try calling `.iter()` or a similar method
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 error: the trait `core::iter::Iterator` is not implemented for the type `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` [E0277]
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 help: run `rustc --explain E0277` to see a detailed explanation
browser.rs:230:9: 244:10 note: `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` is not an iterator; maybe try calling `.iter()` or a similar method
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 error: the trait `core::iter::Iterator` is not implemented for the type `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` [E0277]
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:230:9: 244:10 help: run `rustc --explain E0277` to see a detailed explanation
browser.rs:230:9: 244:10 note: `&core::cell::Ref<'_, collections::vec::Vec<interfaces::cef_browser::CefBrowser>>` is not an iterator; maybe try calling `.iter()` or a similar method
browser.rs:230         for browser in &browsers.borrow() {
browser.rs:231             if browser.downcast().callback_executed.get() == false {
browser.rs:232                 browser_callback_after_created(browser.clone());
browser.rs:233             }
browser.rs:234             let mut events = match browser.downcast().window {
browser.rs:235                 Some(ref win) => win.wait_events(),
               ...
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:235:38: 235:51 error: the type of this value must be known in this context
browser.rs:235                 Some(ref win) => win.wait_events(),
                                                    ^~~~~~~~~~~~~
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
browser.rs:239:22: 239:34 error: the type of this value must be known in this context
browser.rs:239                match events.pop() {
                                    ^~~~~~~~~~~~
note: in expansion of for loop expansion
browser.rs:230:9: 244:10 note: expansion site
note: in expansion of closure expansion
browser.rs:229:19: 245:6 note: expansion site
error: aborting due to 6 previous errors
Could not compile `embedding`.

To learn more, run the command again with --verbose.
CEF build completed in 5.75s
```

**Comment by jdm** [comment]
Huh, we stopped building CEF on our build machines at some point recently apparently.

**Comment by jdm** [comment]
No, that's not true:
http://build.servo.org/builders/linux2/builds/2873/steps/compile_1/logs/stdio is from the most recent automated build.

**Comment by jdm** [comment]
We just stopped recognizing failed builds for it :/

**Comment by jdm** [comment]
Same for android builds, which explains why @connorimes was seeing a build error recently.

**Comment by cb88** [comment]
Ah well good thing I had a go at it manually and reported it then! I was hoping you did have automated builds for CEF... I was thinking of updating an FLTK CEF demo I found.

**Comment by cb88** [comment]
I was able to build-cef. So looks like that is fixed although I'm not sure about the buildbot reporting failures...

**Comment by Ms2ger** [comment]
Yes, sorry, forgot to update this issue. I fixed the build in #7359, and buildbot should report failures now (except for android, which I didn't know how to fix)
